igvc_navigation/src/d_lite_planner/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+add_executable(d_lite_planner
+        ${PROJECT_SOURCE_DIR}/include/${PROJECT_NAME}/Graph.cpp
```

Comment:
I'm confused by the file organization for this target. Why are all of these files in `include`?

Reply:
Because I use them for other planner implementations (i.e. Field D*). Do you think it would be better practice to have `include/src` and place the files in there?

Reply:
"include" is generally for headers; "src" is generally for implementation files (*.cpp).

If you have common code between multiple executables, I'd make a library target in cmake. Give those files their own directory with appropriate "include/*.h" and "src/*.cpp" directories. Then individual executables can include the headers they need from that library and link against the library to get the definitions. 

If the common code is less substantial, you _can_ skip the library target, make the right folder structure, and simply add the common *.cpp files directly to each executable target, but this is arguably less beneficial to humanity.

Reply:
This definitely organized a code. I created a `STATIC` library under `igvc_navigation` and placed relevant files in there:
```
igvc_navigation
|...
|___ GraphSearch
    |___ include
    |   |___"Graph.h","Node.h","PriorityQueue.h"
    |___ src 
        |___"Graph.cpp","Node.cpp","PriorityQueue.cpp"

```
The `d_lite_planner` executable now links this library